### PR TITLE
Add support for latest qwopqwop200 cuda branch

### PR DESCRIPTION
**Pull request by oobabooga**
Following the discussion in #734

### CODE REVIEW

**Comment by martintops** [comment]
Not sure this is related or just due to qwopqwop200's changes, but this seems to break multi-gpu with the latest version of qwopqwop200 cuda branch.

Running the same command `python server.py --model vicuna-13b --gpu-memory 5 7 --wbits 4 --groupsize 128` with the `anon8231489123/vicuna-13b-GPTQ-4bit-128g` model works on your GPTQ fork, but doesn't anymore with qwopqwop200's branch and your fixes here.

Stacktrace:
```
To create a public link, set `share=True` in `launch()`.
Traceback (most recent call last):
  File "/home/user/text-generation-webui/modules/callbacks.py", line 64, in gentask
    ret = self.mfunc(callback=_callback, **self.kwargs)
  File "/home/user/text-generation-webui/modules/text_generation.py", line 224, in generate_with_callback
    shared.model.generate(**kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/torch/utils/_contextlib.py", line 115, in decorate_context
    return func(*args, **kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/transformers/generation/utils.py", line 1416, in generate
    return self.greedy_search(
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/transformers/generation/utils.py", line 2211, in greedy_search
    outputs = self(
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 765, in forward
    outputs = self.model(
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 614, in forward
    layer_outputs = decoder_layer(
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 309, in forward
    hidden_states, self_attn_weights, present_key_value = self.self_attn(
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1501, in _call_impl
    return forward_call(*args, **kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/accelerate/hooks.py", line 165, in new_forward
    output = old_forward(*args, **kwargs)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 219, in forward
    query_states, key_states = apply_rotary_pos_emb(query_states, key_states, cos, sin, offset=offset)
  File "/home/user/miniconda3/envs/textgen/lib/python3.10/site-packages/transformers/models/llama/modeling_llama.py", line 132, in apply_rotary_pos_emb
    q_embed = (q * cos) + (rotate_half(q) * sin)
RuntimeError: Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cuda:1!
```

**Comment by oobabooga** [comment]
I'm inclined to not merge these changes and instead add optional support for the triton branch somehow.

File: modules/GPTQ_loader.py [hunk 1]
```diff
@@ -33,16 +33,16 @@ def noop(*args, **kwargs):
     for name in exclude_layers:
         if name in layers:
             del layers[name]
-    make_quant(model, layers, wbits, groupsize, faster=faster_kernel, kernel_switch_threshold=kernel_switch_threshold)
+    make_quant(model, layers, wbits, groupsize)
```
Comment (EyeDeck) [inline modules/GPTQ_loader.py:36]: We could conditionalize this instead
```python
import inspect
[...]
    gptq_args = inspect.getfullargspec(make_quant).args

    make_quant_kwargs = {
        'module': model, 
        'names': layers,
        'bits': wbits,
    }
    if 'groupsize' in gptq_args:
        make_quant_kwargs['groupsize'] = groupsize
    if 'faster' in gptq_args:
        make_quant_kwargs['faster'] = faster_kernel
    if 'kernel_switch_threshold' in gptq_args:
        make_quant_kwargs['kernel_switch_threshold'] = kernel_switch_threshold
    
    make_quant(**make_quant_kwargs)
```
I think that also makes this backwards-compatible with the pre-groupsize GPTQ-for-LLaMA.
The same thing can be repeated for `load_quant` below.

Comment (oobabooga) [reply modules/GPTQ_loader.py:36]: Very nice! I'll use your code.

Comment (EyeDeck) [reply modules/GPTQ_loader.py:36]: I can make a proper pull request shortly, just had to wait on a quantization for triton first for some testing. It does seem to work fine on GPTQ-for-LLaMA old CUDA/new CUDA/triton branches without having to tweak anything.

Comment (oobabooga) [reply modules/GPTQ_loader.py:36]: I would appreciate a PR. I'm refactoring several things in the interface and haven't been testing GPTQ much.

**Pull request closed without merging**